src/order/compactly_generated.lean

```diff
@@ -247,6 +250,15 @@ begin
   exact le_Sup hx₀,
 end
 
+lemma well_founded.finite_of_independent (hwf : well_founded ((>) : α → α → Prop))
+  {ι : Type*} {t : ι → α} (ht : independent t) (h_ne_bot : ∀ i, t i ≠ ⊥) : finite ι :=
+begin
+  suffices : (set.range t).finite,
+  { haveI : finite (set.range t) := this.finite,
+    exact finite.of_equiv (set.range t) (equiv.of_injective _ (ht.injective h_ne_bot)).symm, },
```

Comment:
Can you use `finite.of_injective` here?

Reply:
Hmm, not directly because the codomain is not finite. I guess there is a lemma to break out here: an injective function with finite range must have finite domain. I'll add that.